app/src/main/java/org/xbmc/kore/ui/sections/remote/OpenSharedUrl.java

```diff
@@ -71,37 +75,46 @@ public Boolean call() throws Error, InterruptedException {
             List<PlayerType.GetActivePlayersReturnType> players =
                     hostConnection.execute(new Player.GetActivePlayers())
                         .get();
-            boolean videoIsPlaying = false;
+            boolean mediaIsPlaying = false;
             for (PlayerType.GetActivePlayersReturnType player : players) {
                 if (player.type.equals(PlayerType.GetActivePlayersReturnType.VIDEO)) {
-                    videoIsPlaying = true;
+                    mediaIsPlaying = true;
                     break;
                 }
             }
 
             stage = R.string.error_queue_media_file;
-            if (!videoIsPlaying) {
-                LogUtils.LOGD(TAG, "Clearing video playlist");
-                hostConnection.execute(new Playlist.Clear(PlaylistType.VIDEO_PLAYLISTID))
+            if (!mediaIsPlaying) {
+                LogUtils.LOGD(TAG, "Clearing playlist number " + playlistType);
+                hostConnection.execute(new Playlist.Clear(playlistType))
                     .get();
             }
 
-            LogUtils.LOGD(TAG, "Queueing file");
-            PlaylistType.Item item = new PlaylistType.Item();
-            item.file = pluginUrl;
-            hostConnection.execute(new Playlist.Add(PlaylistType.VIDEO_PLAYLISTID, item))
-                .get();
+            if (queue) {
```

Comment:
[related to the other comment]

AFAICT the previous behaviour is similar to having `queue` set to `true` so that should be the default in `handleStartIntent`